assignments/prototypes.js

```diff
@@ -83,28 +150,114 @@
 
   // Stretch task: 
   // * Create Villain and Hero constructor functions that inherit from the Humanoid constructor function.  
   // * Give the Hero and Villains different methods that could be used to remove health points from objects which could result in destruction if health gets to 0 or drops below 0;
-  // * Create two new objects, one a villain and one a hero and fight it out with methods!
\ No newline at end of file
+  // * Create two new objects, one a villain and one a hero and fight it out with methods!
+
+  function hero(heroattributes) {
+    //This binds the "this" keyword to CharacterStats
+    Humanoid.call(this, heroattributes);
+    this.heropower = heroattributes.heropower;
+  }
+  
+  // this sets up the __proto__ and allows us to use methods now across objects (destroy & takeDamage)
+  hero.prototype = Object.create(Humanoid.prototype);
+  
+  hero.prototype.discover = function () {
+    return (`${this.name} discovers ${this.heropower}.`)
+  }
+  
+  hero.prototype.heal = function (character, hpheal) {
+    console.log(`${this.name} has healed ${character.name}!. ${character.name}! gains ${hpheal} points.`);
+    character.hp = character.hp + hpheal;
+    return (character.hp);
+  }
+  
+  const FlameWarrior = new hero({
+    createdAt: new Date(),
+    dimensions: {
+      length: 1,
+      width: 2,
+      height: 5,
+    },
+    hp: 20,
+    name: 'Ryo Sanada',
+    faction: 'Ronin Warriors',
+    weapons: [
+      'Sword',
+      'White Blaze',
+    ],
+    language: 'Japanese',
+  });
+  
+  console.log(FlameWarrior.heal(swordsman, 5));
+  
+  function villain(villainattributes) {
+    //This binds the "this" keyword to CharacterStats
+    Humanoid.call(this, villainattributes);
+    this.defend = villainattributes.defend;
+  }
+  
+  // this sets up the __proto__ and allows us to use methods now across objects (destroy & takeDamage)
+  villain.prototype = Object.create(Humanoid.prototype);
+  
+  villain.prototype.taunt = function () {
+    return (`Loser, your powers are too weak to defeat ${this.name}!`)
+  }
+  
+  villain.prototype.curse = function(character, hpcurse) {
+    character.hp = character.hp - hpcurse;
+    if (character.hp <= 0) {
+      console.log(`${this.name} has cursed ${character.name}!. ${character.name}! loses ${hpcurse} points.`);
+      console.log(character.destroy()); 
+      return (character.hp);
+    }
+    else {
+      console.log(`${this.name} has cursed ${character.name}!. ${character.name}! loses ${hpcurse} points.`);
+       return (character.hp);
+    }
+  }
+  
+  const Talpa = new villain({
+    createdAt: new Date(),
+    dimensions: {
+      length: 1,
+      width: 2,
+      height: 3,
+    },
+    hp: 20,
+    name: 'Talpa, The Dark Emperor',
+    faction: 'Nether Realm',
+    weapons: [
+      'Double-Edged Swords',
+      'Dark Warlords',
+    ],
+    language: 'Ancient',
+  });
+  
+  console.log(Talpa.curse(mage, 25));
```

Comment:
These 2 logs will return as NaN (not a number), have a look at them and see if you can debug this problem.